### PR TITLE
test(custom_profile_d): Longevity with custom_d profile

### PR DESCRIPTION
**Pull request by aleksbykov**
Add new longevity with specific custom payload
 which allow to reproduce issue scylladb/scylladb#12562
    
Specific keyspace defined in c-s profile yamls.
C-s run payload run under specific user. Specific
user is added using sla utils. If scylla is OSS, then
just roles will be added and c-s will run with defined
role, if scylla is enterprise, then  full WLP will
be used.

task: https://github.com/scylladb/qa-tasks/issues/974

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
Job in progress

File: sdcm/sla/libs/sla_utils.py
Comment (roydahan) [inline sdcm/sla/libs/sla_utils.py:341]: Don't we have create_role somewhere??

File: test-cases/longevity/longevity_d_repo.yaml
Comment (roydahan) [inline test-cases/longevity/longevity_d_repo.yaml:24]: let's call it 'longevity-user-profile-d'

**Review by roydahan — CHANGES_REQUESTED** [review]

**Comment by roydahan** [comment]
Also, let's wait for the PR from @soyacz that adds multi-AZ and include it here as well.

File: test-cases/longevity/longevity_d_repo.yaml
Comment (roydahan) [inline test-cases/longevity/longevity_d_repo.yaml:20]: We want to have nemesis here at some point.
It's ok to start without for initial testing to make sure it behaves as we expect, but then (before merge) let's change to Sisyphus and test it.

**Comment by aleksbykov** [comment]
https://github.com/scylladb/qa-tasks/issues/974

**Comment by aleksbykov** [comment]
The only 1 unit-test failed: unit_tests/test_sct_events_file_logger.py::TestFileLogger::test_get_events_grouped_by_category_limit and was related to this PR

File: sdcm/sla/libs/sla_utils.py
Comment (aleksbykov) [reply sdcm/sla/libs/sla_utils.py]: used sla_utils to create new roles

File: sdcm/tester.py
Comment (aleksbykov) [reply sdcm/tester.py]: Rewrote to use sla utils for role creation

File: test-cases/longevity/longevity_d_repo.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity_d_repo.yaml]: decrease loader type and increase loader number

File: test-cases/longevity/longevity_d_repo.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity_d_repo.yaml]: yep, i fixed command. now it take about 500GB. I run additional job with increased prepare to get total size.

File: test-cases/longevity/longevity_d_repo.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity_d_repo.yaml]: removed

File: test-cases/longevity/longevity_d_repo.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity_d_repo.yaml]: Added

**Comment by aleksbykov** [comment]
JObs: 
- https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/d-repo-staging/45
- https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/d-repo-staging/48

File: data_dir/profile-cache.yaml
Comment (roydahan) [inline data_dir/profile-cache.yaml:1]: Move it out of here to scylla-qa-internal

File: data_dir/profile-cold-read.yaml
Comment (roydahan) [inline data_dir/profile-cold-read.yaml:1]: Same as above

File: data_dir/profile-top.yaml
Comment (roydahan) [inline data_dir/profile-top.yaml:1]: Same

File: data_dir/profile-write.yaml
Comment (roydahan) [inline data_dir/profile-write.yaml:1]: Same

**Comment by aleksbykov** [comment]
@roydahan could you merge pr: https://github.com/scylladb/scylla-qa-internal/pull/29

File: data_dir/profile-cache.yaml
Comment (aleksbykov) [reply data_dir/profile-cache.yaml:1]: moved

File: data_dir/profile-cold-read.yaml
Comment (aleksbykov) [reply data_dir/profile-cold-read.yaml:1]: moved

File: data_dir/profile-top.yaml
Comment (aleksbykov) [reply data_dir/profile-top.yaml:1]: moved

File: data_dir/profile-write.yaml
Comment (aleksbykov) [reply data_dir/profile-write.yaml:1]: moved

**Comment by aleksbykov** [comment]
@roydahan @vponomaryov @fruch @soyacz @temichus Could you review.

**Comment by aleksbykov** [comment]
@roydahan 
i fixed according your comments. Also add some fixes to sct. 
I create job: https://jenkins.scylladb.com/view/master/job/scylla-master/job/longevity/job/longevity-user-profile-d-4h-test/ for master. with master issue was not reproduced, while have similar pictures on dashboards but latencies are small. With 2022.2 and 2022.1 the issue reproduced.
Job for master is passed. For now it is use only single user to send traffic.  If we want, for enterprise we can add second users and set Shares for operations.

@roydahan @fruch @juliayakovlev @soyacz @vponomaryov could you please review

File: test-cases/longevity/longevity-user-profile-d.yaml
Comment (juliayakovlev) [inline test-cases/longevity/longevity-user-profile-d.yaml:3]: Maximum allowed service level value is 1000. Do you see that service_level with `2000` shares is created? If yes - it looks like an issue

File: data_dir/profile-cache.yaml
Comment (juliayakovlev) [reply data_dir/profile-cache.yaml:1]: @roydahan why it should be placed under scylla-qa-internal? How we should decide where to place user profile?

**Comment by juliayakovlev** [comment]
What does `user-profile-d` name mean? Disney?

File: test-cases/longevity/longevity-user-profile-d.yaml
Comment (aleksbykov) [reply test-cases/longevity/longevity-user-profile-d.yaml:3]: In this job , it is mostly used as role name, because on master there is no SLA. 
But i will change it to 1000, so it also could run on enterprise

File: longevity_sla_test.py
Comment (juliayakovlev) [inline longevity_sla_test.py:51]: Why we check it in the enterprise only? Service level is allowed to be created in OSS also. Or am I wrong?

**Review by juliayakovlev — COMMENTED** [review]
small question

File: longevity_sla_test.py
Comment (aleksbykov) [reply longevity_sla_test.py:51]: I though that SLA(WLP) is only Enterprise feature.

Comment (aleksbykov) [reply longevity_sla_test.py:51]: https://opensource.docs.scylladb.com/stable/using-scylla/workload-prioritization.html

Comment (juliayakovlev) [reply longevity_sla_test.py:51]: Service level can be created in OSS, just `shares` parameter is not supported. 
But because of we check shares in "wait for propagation" function, this validation should not be used in OSS.

**Review by juliayakovlev — APPROVED** [review]
I stil do not understand what is it the name "custom_d". It's very strange and not understandable

**Review by roydahan — APPROVED** [review]

**Pull request merged**
